components/camel-ai/camel-langchain/src/main/java/org/apache/camel/component/langchain/LangchainEndpoint.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.langchain;
+
+import dev.langchain4j.data.message.ChatMessageType;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import org.apache.camel.Category;
+import org.apache.camel.Consumer;
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriPath;
+import org.apache.camel.support.DefaultEndpoint;
+import org.apache.camel.util.ObjectHelper;
+
+import static org.apache.camel.component.langchain.Langchain4jConstants.ENDPOINT_TYPE_CHAT;
+
+@UriEndpoint(firstVersion = "4.5.0", scheme = "langchain",
+             title = "langchain",
+             syntax = "langchain:type", producerOnly = true,
+             category = { Category.AI })
+public class LangchainEndpoint extends DefaultEndpoint {
+
+    @UriPath(label = "Langchain Endpoint type. There's only one type available today which is the chat",
+             enums = "" + ENDPOINT_TYPE_CHAT)
+    @Metadata(required = true)
+    private String type;
+
+    @UriParam
+    @Metadata(required = true, defaultValue = "CHAT_SINGLE_MESSAGE")
+    private LangchainChatOperations chatOperation = LangchainChatOperations.CHAT_SINGLE_MESSAGE;
+
+    @UriParam(label = "advanced")
+    @Metadata(autowired = true)
+    private ChatLanguageModel chatModel;
+
+    @UriParam
+    private ChatMessageType chatMessageType;
```

Comment:
It would be nice to put the configuration option in a Configuration class so that the options can be set also at component level (like https://github.com/apache/camel/pull/13341/files#diff-bc52e1e75950e211246e45748de59f511718780ff31dbf656c2995949203b5ffR28)